### PR TITLE
chore: add github action to verify code

### LINKED ISSUE #253: enable verify in makefile and our github workflow
We actually have many scripts to verify/test our code but not used yet which is really  great loss here.

The very simple way to enable `verify` is to run `hack/verify-all.sh` but this needs to be verified.
```makefile
.PHONY: verify
verify:
	@hack/verify-all.sh
```

Besides, some scripts should not be used or optimzied. Overrall, this will be a `difficulty-medium` task. 

To enable this in our github workflow, we need to add another action or task.

### PR DESCRIPTION
**Pull request by zqq454224016**
#### Which issue(s) this PR fixes

Fixes #253

### CODE REVIEW

File: config/samples/example-test.sh [hunk 1]
```diff
@@ -537,29 +537,6 @@ info "5.9.3 Plan a nignx with private chartmuseum(basic auth enabled) "
 kubectl apply -f config/samples/core_v1alpha1_componentplan_mynginx.yaml
 waitComponentPlanDone "kubebb-system" "mynginx" 
 
-info "5.8 Verify that helm add repo with basic auth"
```
Comment (Abirdcfly) [inline config/samples/example-test.sh:540]: mabe you can try `git fetch upstream && git checkout upstream/main -- config/sample/example-test.sh` to restore this file.

**Review by Abirdcfly — CHANGES_REQUESTED** [review]

**Comment by Abirdcfly** [comment]
example-test failed reported by #274, maybe fixed by #279

File: config/samples/example-test.sh (hunk 1 above)
Comment (Abirdcfly) [reply config/samples/example-test.sh:540]: +1

Comment (bjwswang) [reply config/samples/example-test.sh:540]: @zqq454224016 pls revert changes in this file as I will fix this issue in another PR.

Comment (Abirdcfly) [reply config/samples/example-test.sh:540]: These deletions didn't actually skip the #274 ...

**Pull request closed without merging**